### PR TITLE
Initial work on the seperation of user form

### PR DESCRIPTION
**Pull request by Mouados-byte**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -61,13 +67,24 @@ class User < ApplicationRecord
   has_many :reviews, through: :review_users
   has_many :family_members, dependent: :destroy
 
+  validates :payment_method, :bank, :base_salary,
+    :retirement_contribution, :rib, presence: true,
```
Comment (mohamedrez) [inline app/models/user.rb:71]: no need to validate presence for these, presence is only for email, first name and last name

File: Gemfile [hunk 2]
```diff
@@ -109,3 +109,5 @@ end
 gem "chartkick", "~> 5.0"
 
 gem "paper_trail"
+
+gem "wicked"
```
Comment (mohamedrez) [inline Gemfile:113]: I don't think we need an external Gem just for this task, I can explain why we should minimise using gems

File: app/models/user.rb [hunk 3]
```diff
@@ -108,6 +125,12 @@ def has_any_role?(role_names = nil)
     end
   end
 
+  def email_uniqueness_if_not_nil
```
Comment (mohamedrez) [inline app/models/user.rb:128]: not needed

**Review by mohamedrez — CHANGES_REQUESTED** [review]
image

We need to disable or hide the tabs if the user does not yet exists

File: Gemfile (hunk 2 above)
Comment (Mouados-byte) [reply Gemfile:113]: okay , imma try working with it

**Pull request closed without merging**